src/transactionIntentsFactories/delegationTransactionIntentsFactory.ts

```diff
@@ -0,0 +1,344 @@
+    additionalGasLimitForDelegationOperations: BigNumber.Value;
+}
+
+interface ValidatorPublicKey {
+    hex(): string;
+}
+
+export class DelegationTransactionIntentsFactory {
+    private readonly config: Config;
+
+    constructor(config: Config) {
+        this.config = config;
+    }
+
+    createTransactionIntentForNewDelegationContract(options: {
+        sender: IAddress,
+        totalDelegationCap: BigNumber.Value,
+        serviceFee: BigNumber.Value,
+        value: BigNumber.Value
+    }): TransactionIntent {
+        const dataParts = [
+            "createNewDelegationContract",
+            numberToPaddedHex(options.totalDelegationCap.toString()),
+            numberToPaddedHex(options.serviceFee.toString())
+        ];
+
+        const executionGasLimit = new BigNumber(this.config.gasLimitCreateDelegationContract).plus(this.config.additionalGasLimitForDelegationOperations);
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: Address.fromBech32(DELEGATION_MANAGER_SC_ADDRESS),
+            dataParts: dataParts,
+            executionGasLimit: executionGasLimit,
+            value: options.value
+        }).build();
+    }
+
+    createTransactionIntentForAddingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[],
+        signedMessages: Uint8Array[]
+    }): TransactionIntent {
+        if (options.publicKeys.length !== options.signedMessages.length) {
+            throw new Err("The number of public keys should match the number of signed messages");
+        }
+
+        const numNodes = options.publicKeys.length;
+
+        let dataParts = ["addNodes"];
+        for (let i = 0; i < numNodes; i++) {
+            dataParts = dataParts.concat(options.publicKeys[i].hex());
+            dataParts = dataParts.concat(byteArrayToHex(options.signedMessages[i]));
+        }
+
+        return new TransactionIntentBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.delegationContract,
+            dataParts: dataParts,
+            executionGasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes)
+        }).build();
+    }
+
+    private computeExecutionGasLimitForNodesManagement(numNodes: number): BigNumber.Value {
+        const additionalGasForAllNodes = new BigNumber(this.config.additionalGasLimitPerValidatorNode).multipliedBy(numNodes);
+        return new BigNumber(this.config.gasLimitDelegationOperations).plus(additionalGasForAllNodes);
+    }
+
+    createTransactionIntentForRemovingNodes(options: {
+        sender: IAddress,
+        delegationContract: IAddress,
+        publicKeys: ValidatorPublicKey[]
+    }): TransactionIntent {
+        let dataParts = ["removeNodes"];
+
+        for (const key of options.publicKeys) {
+            dataParts = dataParts.concat(key.hex());
```

Comment:
I am not sure why `concat` works fine in this case. Does the compiler complain if you add the typing markers `dataParts: string[] = ...`? Perhaps use `append`, as mentioned in the comment above?

Reply:
Now using `push()`. It does not complain if I add typing markers but it already shows the correct type.